src/averageAges.js

```diff
@@ -55,7 +65,14 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(child => people.find(mother => mother.name === child.mother)
+    && child.sex === 'm')
+    : people.filter(child => people.find(mother => mother.name
+      === child.mother));
+
+  return children.reduce((a, b) => a + b.born - people.find(mother => (
+    b.mother === mother.name)).born, 0) / children.length;
```

Comment:
[checklist point 2](https://github.com/mate-academy/js_average-ages/blob/master/checklist.md)

And change names `a` and `b`.